widget/table.go

```diff
@@ -1388,6 +1384,31 @@ func (r *tableCellsRenderer) refreshForID(toDraw TableCellID) {
 	r.hover.Refresh()
 }
 
+func (r *tableCellsRenderer) updateCells(toDraw TableCellID, visible, wasVisible map[TableCellID]fyne.CanvasObject) {
+	updateCell := r.cells.t.UpdateCell
+	if updateCell == nil {
+		fyne.LogError("Missing UpdateCell callback required for Table", nil)
+		return
+	}
+
+	if toDraw == onlyNewTableCellsID {
+		for id, cell := range visible {
+			if _, ok := wasVisible[id]; !ok {
+				updateCell(id, cell)
+			}
+		}
```

Comment:
I think it would be cleaner with an early return here to avoid two indented code blocks.